src/pages/index.js

```diff
@@ -54,31 +91,216 @@ variables.profileEditButton.addEventListener("click", () => {
+/*                  Pass form inputs to validate *Avatar* form data           */
+/* -------------------------------------------------------------------------- */
+
+const avatarFormValidator = new FormValidator(
+  formConfig,
+  variables.avatarModalForm
+);
+avatarFormValidator.enableValidation();
+
+/* -------------------------------------------------------------------------- */
+/*                         Call API and pass info to update the avatar        */
+/* -------------------------------------------------------------------------- */
+
+const avatarPopupForm = new PopupWithForm(
+  "#edit-avatar",
+  ({ link: avatar }) => {
+    variables.avatarModalSaveButton.textContent = "Saving...";
+    api
+      .updateUserAvatar({ avatar })
+      .then((res) => {
+        user.updateAvatar(res.avatar);
+        avatarPopupForm.close();
+        variables.avatarModalForm.reset();
+      })
+      .catch((error) => {
+        console.error("Failed to update user avatar:", error);
+      })
+      .finally(() => {
+        variables.avatarModalSaveButton.textContent = "Save";
+        avatarFormValidator.disableSubmitButton();
+      });
+  }
 );
 
-function renderCard(item) {
-  const cardElement = createCard(item);
-  cardSection.addItem(cardElement);
+/* -------------------------------------------------------------------------- */
+/*                   Edit Avatar Form Event Listener                          */
+/* -------------------------------------------------------------------------- */
+
+avatarPopupForm.setEventListeners();
+variables.avatarChangeBut.addEventListener("click", () => {
+  avatarPopupForm.open();
+});
+
+/* -------------------------------------------------------------------------- */
+/*                  Create card object and render the images                  */
+/* -------------------------------------------------------------------------- */
+
+function createCard(card) {
+  const newCard = new Card(
+    card,
+    "#elementCard",
+    handleImageClick,
+    handleDeleteCardButtonClick,
+    likeCardAPI,
+    unlikeCardAPI
+  );
+
+  return newCard.getView();
 }
 
-cardSection.renderItems();
+/* -------------------------------------------------------------------------- */
+/*              Handle like/dislike functionality callback                    */
+/* -------------------------------------------------------------------------- */
+
+const likeCardAPI = (cardData, changeLike) => {
+  api
+    .likeCard(cardData)
+    .then(() => {
+      changeLike.classList.toggle("element__like-button_active");
+      console.log("Card liked successfully.");
+    })
+    .catch((error) => {
+      console.error("Failed to like card:", error);
+    });
+};
+const unlikeCardAPI = (cardData, changeLike) => {
+  api
+    .unlikeCard(cardData)
+    .then(() => {
+      changeLike.classList.toggle("element__like-button_active");
+      console.log("Card disliked successfully.");
+    })
```

Comment:
Now only class `Card` knows every **card selector**, so you need to encapsulate the likes logics inside its methods and call them in `index.js`. It’s a violation of **Single responsibility** principle


Please, make method `toggleLike` and call it here

```js
const likeCardAPI = (cardData, cardInstance) => {
  api
    .likeCard(cardData)
    .then(() => {
      cardInstance.toggleLike();
    })
    .catch((error) => {
      console.error("Failed to like card:", error);
    });
};
const unlikeCardAPI = (cardData, cardInstance) => {
  api
    .unlikeCard(cardData)
    .then(() => {
      cardInstance.toggleLike();
    })
    .catch((error) => {
      console.error("Failed to unlike card:", error);
    });
};

```